src/main/java/io/confluent/connect/jdbc/source/TableMonitorThread.java

```diff
@@ -80,74 +85,70 @@ public void run() {
     }
   }
 
-  public List<String> tables() {
+  public synchronized List<String> tables() {
     //TODO: Timeout should probably be user-configurable or class-level constant
     final long timeout = 10000L;
-    synchronized (db) {
-      long started = System.currentTimeMillis();
-      long now = started;
-      while (tables == null && now - started < timeout) {
-        try {
-          db.wait(timeout - (now - started));
-        } catch (InterruptedException e) {
-          // Ignore
-        }
-        now = System.currentTimeMillis();
-      }
-      if (tables == null) {
-        throw new ConnectException("Tables could not be updated quickly enough.");
+    long started = System.currentTimeMillis();
+    long now = started;
+    while (tables == null && now - started < timeout) {
+      try {
+        wait(timeout - (now - started));
+      } catch (InterruptedException e) {
+        // Ignore
       }
-      return tables;
+      now = System.currentTimeMillis();
+    }
+    if (tables == null) {
+      throw new ConnectException("Tables could not be updated quickly enough.");
     }
+    return tables;
   }
 
   public void shutdown() {
     shutdownLatch.countDown();
   }
 
   // Update tables and return true if the
-  private boolean updateTables() {
-    synchronized (db) {
-      final List<String> tables;
-      try {
-        tables = JdbcUtils.getTables(db, schemaPattern, tableTypes);
-        log.debug("Got the following tables: " + Arrays.toString(tables.toArray()));
-      } catch (SQLException e) {
-        log.error("Error while trying to get updated table list, ignoring and waiting for next "
-                  + "table poll interval", e);
-        return false;
-      }
+  private synchronized boolean updateTables() {
```

Comment:
Unrelated, but the comment is incomplete (which is almost definitely my fault -- I have a tendency to leave half completed sentences...)